content/docs/uncontrolled-components.md

```diff
@@ -6,28 +6,29 @@ permalink: docs/uncontrolled-components.html
 
 In most cases, we recommend using [controlled components](/docs/forms.html) to implement forms. In a controlled component, form data is handled by a React component. The alternative is uncontrolled components, where form data is handled by the DOM itself.
 
-To write an uncontrolled component, instead of writing an event handler for every state update, you can [use a ref](/docs/refs-and-the-dom.html) to get form values from the DOM.
+To write an uncontrolled component, instead of writing an event handler for every state update, you can read form values from the [form](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement#Properties) itself.
 
 For example, this code accepts a single name in an uncontrolled component:
 
-```javascript{8,17}
+```javascript{9,10,18}
 class NameForm extends React.Component {
   constructor(props) {
     super(props);
     this.handleSubmit = this.handleSubmit.bind(this);
   }
 
   handleSubmit(event) {
-    alert('A name was submitted: ' + this.input.value);
     event.preventDefault();
+    const name = event.target.elements.name.value;
```

Comment:
Perhaps this can be expanded to a couple of lines to make it clearer for everyone:

```javascript
const form = event.target;
const nameInput = form.elements.name;
const name = nameInput.value;
```